### PR TITLE
Add Pomerium Zero Helm chart

### PR DESCRIPTION
**Pull request by desimone**
- Create initial Helm chart for deploying Pomerium Zero
- Implement basic deployment, service, and secret resources
- Add configurable values for token, image, and service settings
- Include health check test for /healthz endpoint
- Provide comprehensive README with installation and usage instructions
- Ensure chart follows Helm best practices and naming conventions
- Add Apache 2.0 license

This PR introduces a new Helm chart for easy deployment and management of Pomerium Zero in Kubernetes environments. It includes essential resources, configuration options, and a basic health check test.

## Checklist

- [ ] reference any related issues
- [ ] add appropriate tag (`improvement` / `bug` / etc)
- [ ] ready for review

### CODE REVIEW

File: zero/helm/values.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
+  # limits:
+  #   cpu: 100m
+  #   memory: 128Mi
+  # requests:
+  #   cpu: 100m
+  #   memory: 128Mi
+
+service:
+  type: ClusterIP
```
Comment (wasaga) [inline zero/helm/values.yaml:27]:
```suggestion
  type: LoadBalancer
```

File: zero/helm/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "pomerium-zero.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "pomerium-zero.selectorLabels" . | nindent 8 }}
+    spec:
+      containers:
```
Comment (wasaga) [inline zero/helm/templates/deployment.yaml:18]: can we make it non-root and readonly fs (see https://github.com/pomerium/install/pull/4/files)

Comment (desimone) [reply zero/helm/templates/deployment.yaml:18]: I think this (why only port 80), the additional tmp files, and why/how you are sourcing secrets is a little over my head.

File: zero/helm/values.yaml (hunk 1 above)
Comment (desimone) [reply zero/helm/values.yaml:27]: Good catch. I've updated this and note about pointing DNS.

**Comment by desimone** [comment]
nota bene: The original chart uses `pomerium.io`

https://artifacthub.io/packages/helm/pomerium/pomerium

File: zero/helm/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+type: application
+version: 0.27.2
+appVersion: "v0.27.2"
+home: https://www.pomerium.com/
+icon: https://www.pomerium.com/static-img/favicon.svg
+sources:
+  - https://github.com/pomerium/pomerium
+maintainers:
+  - name: Pomerium Team
+    email: info@pomerium.com
```
Comment (wasaga) [inline zero/helm/Chart.yaml:13]:
```suggestion
    email: support@pomerium.com
```

I think info goes to sales

**Review by gaurdro — APPROVED** [review]

**Review by calebdoxsey — APPROVED** [review]

**Pull request merged**